// @dmmdjs/dmmd.js/dmath/exports.js
"use strict";

// Imports
const miscellaneous = {
    ceil: require("./miscellaneous/ceil.js"),

    compare: require("./miscellaneous/compare.js"),
    
    decimal: require("./miscellaneous/decimal.js"),
    
    floor: require("./miscellaneous/floor.js"),
    
    integer: require("./miscellaneous/integer.js"),
    get int() { return miscellaneous.integer },
    get truncate() { return miscellaneous.integer },
    get trunc() { return miscellaneous.integer },

    invert: require("./miscellaneous/invert.js"),

    isInteger: require("./miscellaneous/isInteger.js"),
    get isInt() { return miscellaneous.isInteger },

    isNegative: require("./miscellaneous/isNegative.js"),
    get isNeg() { return miscellaneous.isNegative },

    isNumber: require("./miscellaneous/isNumber.js"),
    get isNum() { return miscellaneous.isNumber },

    isPositive: require("./miscellaneous/isPositive.js"),
    get isPos() { return miscellaneous.isPositive },

    isZero: require("./miscellaneous/isZero.js"),

    maximum: require("./miscellaneous/maximum.js"),
    get max() { return miscellaneous.maximum },

    minimum: require("./miscellaneous/minimum.js"),
    get min() { return miscellaneous.minimum },

    negative: require("./miscellaneous/negative.js"),
    get neg() { return miscellaneous.negative },

    positive: require("./miscellaneous/positive.js"),
    get pos() { return miscellaneous.positive },
    get absolute() { return miscellaneous.positive },
    get abs() { return miscellaneous.positive },

    round: require("./miscellaneous/round.js"),

    shift: require("./miscellaneous/shift.js"),

    sign: require("./miscellaneous/sign.js")
};
const operators = {
    divide: require("./operators/divide.js"),
    get quotient() { return operators.divide },

    gdivide: require("./operators/gdivide.js"),

    modulo: require("./operators/modulo.js"),
    get modulus() { return operators.modulo },
    get remainder() { return operators.modulo },

    multiply: require("./operators/multiply.js"),
    get product() { return operators.multiply },

    plus: require("./operators/plus.js"),
    get add() { return operators.plus },
    get sum() { return operators.plus },

    subtract: require("./operators/subtract.js"),
    get minus() { return operators.subtract },
    get difference() { return operators.subtract },
    get diff() { return operators.subtract }
};
const utils = {
    format: require("./utils/format.js"),

    isParsed: require("./utils/isParsed.js"),

    parse: require("./utils/parse.js"),

    trim: require("./utils/trim.js")
};

// Exports
module.exports = {
    miscellaneous, ...miscellaneous,
    operators, ...operators,
    utils, ...utils
};
